### DISCUSSION TITLE
Firejail and qutebrwoser installed via virtualenv

### DISCUSSION THREAD

**Discussion by phfqjkml** [qutebrowser/qutebrowser#6427] [Q&A] [answered]
I tried to run qutebrowser with firejail by
`firejail --profile=qutebrowser --whitelist=$PWD/.venv ./.venv/bin/python3 -m qutebrowser`
but what I got is
`No module named qutebrowser`
Any tips on how to make it work?

**Comment by toofar** [comment, marked as answer]
How did you create the virtualenv?

**Reply by phfqjkml** [reply]
`git clone --depth 1 https://github.com/qutebrowser/qutebrowser`
`cd qutebrowser`
`python3 scripts/mkvenv.py --pyqt-type link --skip-docs`

**Reply by toofar** [reply]
Hmm, I suppose you could try getting a python shell invoked the same way and checking sys.path. For my venv (created the same way) what I'm assuming is pointing it to the qutebrowser source is `.venv/lib/python3.9/site-packages/qutebrowser.egg-link`. So making shore that site-packages path is in there would be good.
The main thing that has broken if for me is when upgrading python minor versions I have to recreate the virtualenv (because it uses system packages by default and they move when that happens). Although that probably depends on what distro you are on.

**Reply by phfqjkml** [reply]
sys.path does include `.venv/lib/python3.9/site-packages` and all of the following work
`import adblock`
`import PyQt5`
`import colorama`
Distro is slackware-current.

**Reply by phfqjkml** [reply]
Okay. Checking the content of qutebrowser.egg-link, I find that the directory including qutebrowser should be whitelisted. The problem is resolved!
